### PR TITLE
validate fail missing bracket pair

### LINKED ISSUE #1442: cylc validate: ( foo & bar => baz not fail validation
Examples:
```cfg
[scheduling]
    [[dependencies]]
                    graph = """
                        (foo | bar => baz
                    """
```
fails validation as expected, but
```cfg
[scheduling]
    [[dependencies]]
                    graph = """
                        (foo & bar => baz
                    """
```
does not.

### PR DESCRIPTION
**Pull request by kaday**
Closes #1442

### CODE REVIEW

**Comment by kaday** [comment]
Rebased.

@benfitzpatrick please review

**Comment by kaday** [comment]
Changes made on @matthewrmshin request, back to you @benfitzpatrick

File: lib/cylc/config.py [hunk 1]
```diff
@@ -1309,6 +1309,11 @@ def process_graph_line( self, line, section, seq, offset_seq_map,
             right_nodes = new_right_nodes
 
             # extract task names from lexpression
+            n_open_brackets = len(re.findall(r"(\()", lexpression))
```
Comment (benfitzpatrick) [inline lib/cylc/config.py]: You can do `lexpression.count` here

**Comment by benfitzpatrick** [comment]
Approach looks good

**Comment by kaday** [comment]
Done and tested, back to you @benfitzpatrick

**Pull request merged**